File: Python/Day10.py
from itertools import groupby

number = open('../inputs/Day10.txt').read()

def lookAndSay(number):
	newNumber = ''
	for num, nums in groupby(number):
		newNumber += str(len(list(nums))) + str(num)
	return newNumber

for _ in range(40):
	number = lookAndSay(number)

after40 = len(number)

for _ in range(10):
	number = lookAndSay(number)

after50 = len(number)

print "Part 1:", after40
print "Part 2:", after50

File: JavaScript/Day10.js
(function(){
    "use strict";

    require("fs").readFile("../inputs/Day10.txt", "utf8", function(err, data){
        if(err){
            console.log(`Unable to load file: ${err}`);
        } else {
            let number = data, lookAndSay = num => number.match(/(\d)\1*/g).map(n => n.length + n[0]).join('');

            for(var i = 0; i < 40; i++){
                number = lookAndSay(number);
            }

            let after40 = number.length;

            for(let i = 0; i < 10; i++){
                number = lookAndSay(number);
            }

            let after50 = number.length;

            console.log(`Part 1: ${after40}`);
            console.log(`Part 2: ${after50}`);
        }
    });
})();